keps/sig-storage/20200319-selinux-relabeling.md

```diff
@@ -0,0 +1,359 @@
+- [Motivation](#motivation)
+  - [SELinux intro](#selinux-intro)
+  - [SELinux context assignment](#selinux-context-assignment)
+  - [Volumes](#volumes)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    - [<code>mount -o context</code>](#)
+  - [New Kubernetes behavior](#new-kubernetes-behavior)
+    - [Open Questions](#open-questions)
+  - [Shared volumes](#shared-volumes)
+  - [User Stories [optional]](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+  - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional-1)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+  - [<code>FSGroupChangePolicy</code> approach](#-approach)
+  - [Change container runtime](#change-container-runtime)
+  - [Move SELinux label management to kubelet](#move-selinux-label-management-to-kubelet)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+This KEP tries to speed up the way how volumes (incl. persistent volumes) are made available to Pods on systems with SELinux in enforcing mode.
+Current way includes recursive relabeling of all files on a volume before a container can be started. This is slow for large volumes.
+
+## Motivation
+
+### SELinux intro
+On Linux machines with SELinux in enforcing mode, SELinux tries to prevent users that escaped from a container to access the host OS and also to access other containers running on the host.
+It does so by running each container with unique *SELinux context* (such as `system_u:system_r:container_t:s0:c309,c383`; shortened as `s0:c309,c383` in further text) and labeling all content on all volumes with the same label (`s0:c309,c383`).
```

Comment:
You should state that it expects all content to be labeled system_u:object_r:container_file_t:s0:c309,c383

Reply:
fixed